relayer/core/relay.go

```diff
@@ -186,6 +217,8 @@ func (re *Relay) Start() {
 		log.WithError(ctx.Err()).Error("Goroutines appear deadlocked. Killing process")
 		re.ethChain.Stop()
 		re.subChain.Stop()
+		// re.database.Stop() // TODO: graceful shutdown
```

Comment:
do in this PR plz as it is related to general plumbing/structure of relayer

Reply:
https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/f03c815ec36c601637c07e7ca1695ce85ab6e85a, https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/4a14bcb4c831ad4ecb1aebad47afc2cce0ec6339